_posts/2020-09-21-porting-namecoin-in-tor-browser-to-macos.md

```diff
@@ -0,0 +1,46 @@
+---
+layout: post
+title: "Porting Namecoin in Tor Browser to macOS"
+author: Jeremy Rand
+tags: [News]
+---
+
+Hot on the heels of [porting Namecoin in Tor Browser to Windows]({{ "/2020/08/28/porting-namecoin-in-tor-browser-to-windows.html" | relative_url }}), it was time to do the final remaining desktop OS, macOS.  This one was always going to present a challenge in the QA testing department, because I don't own any macOS devices.  On the other hand, I fully expected that the amount of code written for macOS support was going to be pretty minimal; in fact I expected that other than tweaking a few constants, macOS support was likely to be a subset of the code needed for Windows support.  But of course, our friend Murphy wasn't going to let that assumption stand.
+
+On the bright side, two of our community members who do have macOS devices volunteered to test my binaries and supply me with debug logs when things failed.  So at least that wasn't going to prevent a successful port.
+
+## Initial Attempt
+
+As a first try, I added the Electrum-NMC, ncprop279, and StemNS dependencies (with their config files) to the `tor-browser` rbm project, and added the missing lines in Tor Launcher for determining the paths of Electrum-NMC and StemNS on macOS.  Then I fliped the `var/namecoin` switch for macOS in `tor-browser-build`, and started up a build job.  In order to build without errors, I then needed to make some slight edits to the build scripts of some of the Go projects that `ncprop279` depends on, mostly related to cgo edge cases.
```

Comment:
>depends on

(up)on which ... depends

Reply:
I'm aware that English theoretically has a rule about ending sentences with prepositions, but I find that this rule makes things less pleasant to read, and so I deliberately do not follow it.  (I'm a language anarchist, I have no issue with violating theoretical language rules if doing so makes the writing more effective.)